lib/rules/custom-property-empty-line-before/index.js

```diff
@@ -116,12 +120,15 @@ function rule(expectation, options, context) {
 			});
 		});
 	};
-}
+};
 
+/**
+ * @param {import('postcss').Declaration} decl
+ */
 function isAfterCustomProperty(decl) {
 	const prevNode = getPreviousNonSharedLineCommentNode(decl);
 
-	return prevNode && prevNode.prop && isCustomProperty(prevNode.prop);
+	return prevNode != null && isDeclaration(prevNode) && isCustomProperty(prevNode.prop);
```

Comment:
[note] `getPreviousNonSharedLineCommentNode()` returns a `Node` or `undefined` type, so it's necessary to check the return value via the `isDeclaration()` type guard.

https://github.com/stylelint/stylelint/blob/af4314d1e6b67801e25124ccd1a476ba6d81d8a9/lib/utils/getPreviousNonSharedLineCommentNode.js#L14